### PR TITLE
feat!: updated button styling

### PR DESCRIPTION
**Pull request by jzempel**
- [x] **BREAKING CHANGE?** 

## Description

For visual parity with https://garden.zendesk.com/react-components/buttons/. Breaking changes include:
- removal of `.c-btn--muted`; use `.c-btn--icon.c-btn--basic` instead
- removal of `.l-btn-group`; use react-components [`<ButtonGroup>`](https://garden.zendesk.com/react-components/buttons/#buttongroup) or [`<SplitButton>`](https://garden.zendesk.com/react-components/buttons/#splitbutton) for required accessibility logic
- removal of `custom.css`; Garden css-components will no longer bundle custom color demos – use react-components and [Theming](https://garden.zendesk.com/react-components/theming/) when custom styling is needed

Features/improvements include:
- added support for [`:focus-visible` polyfill](https://github.com/WICG/focus-visible); demonstrated in the example page
- applied [react-components API](https://github.com/zendeskgarden/react-components/blob/master/docs/api.md#view-components) property ordering best-practice for simplified specificity and drastically reduced size.

## Checklist

* [x] :ok_hand: style updates are Garden Designer approved (add the
  designer as a reviewer)
* [x] :globe_with_meridians: component demo is up-to-date (`yarn start`)
* [x] :white_check_mark: all component states are represented
  (`.is-hovered`, `.is-focused`, etc.)
* [x] :arrow_left: renders as expected with reversed (RTL) direction
* [x] :metal: renders as expected sans Bedrock (`?bedrock=false`)
* [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,9 @@
     "tag": "lerna version --conventional-commits",
     "test": "lerna run test --stream"
   },
+  "dependencies": {
+    "focus-visible": "^5.1.0"
```
Comment (austin94) [inline package.json:17]: Should this be a dev dependency? Consumers wouldn't need this in their `node_modules` to consume CSS.

File: packages/buttons/src/_selectors.css [hunk 2]
```diff
@@ -1,22 +1,15 @@
...
-  .c-btn--primary,
-  .is-selected;
-
-@custom-selector :--btn-hovered
-  .is-hovered.is-hovered,
-  :hover:hover;
-
-@custom-selector :--btn-focused
+@custom-selector :--btn-focus
+  :focus-visible,
```
Comment (austin94) [inline packages/buttons/src/_selectors.css:6]: With [less than 5% of browsers supporting this](https://caniuse.com/#search=focus-visible) and the [proposal still being a working draft](https://drafts.csswg.org/selectors-4/#the-focus-visible-pseudo) I'm not sure it's the right time to add this. Requiring consumers to include an additional polyfill, even if it is official, seems at odds with providing native CSS.

File: packages/buttons/src/_group.css [hunk 3]
```diff
@@ -1,137 +0,0 @@
...
-
-:root {
-  --zd-btn-group--light-border-color: var(--zd-color-grey-100);
-  --zd-btn-group--primary-focused-box-shadow:
-    inset 0 0 0 3px
-    color-mod(var(--zd-color-white) alpha(35%));
-  --zd-btn-group--primary-line-height: var(--zd-btn--icon-size);
-  --zd-btn-group--primary--sm-line-height: var(--zd-btn--icon--sm-size);
-  --zd-btn-group--primary--lg-line-height: var(--zd-btn--icon--lg-size);
-}
```
Comment (austin94) [inline packages/buttons/src/_group.css:12]: Could you go into detail the `group` styling removal? You mentioned

> for required accessibility logic

but, I think the only required logic is `role="group"` on the containing element.

File: package.json (hunk 1 above)
Comment (austin94) [reply package.json:17]: Just realized this is in the project root. This should also be pinned as well.

Comment (jzempel) [reply package.json]: oops, did a blind `yarn add`. Fixed.

File: packages/buttons/src/_selectors.css (hunk 2 above)
Comment (jzempel) [reply packages/buttons/src/_selectors.css:6]: This is a more mainstream option than the previous `.is-focused`. Now both are supported and folks don't need to roll their own custom JS to achieve the right focus/click behavior.

File: packages/buttons/src/_group.css (hunk 3 above)
Comment (jzempel) [reply packages/buttons/src/_group.css:12]: Clearly there is more to the React component than `role`. See https://github.com/zendeskgarden/react-components/blob/master/packages/buttons/src/elements/ButtonGroup.tsx for controlling selection and default selection. We don't want to carry that weight (JS required) over here.

Comment (austin94) [reply packages/buttons/src/_group.css:12]: > Clearly there is more to the React component than role

Agreed. But, those additions, including roving-tab-index, aren't required a11y. Only `[role="group", aria-label="some label"]` are required from my understanding.

> We don't want to carry that weight over here

Agreed. We historically haven't supported those interactions in this package, I don't think now is the time to make that change.

**Pull request merged**